tests/client/test_query.py

```diff
@@ -55,3 +56,65 @@ def test_get_container_sources():
         'upstream_urls': [yum_utils_upstream_url, etcd_upstream_url]
     }
     assert query.get_container_content_sources(container_build_id) == expected
+
+
+def test_get_source_components_in_archive():
+    """Test the get_source_components_in_archive function."""
+
+    # Set up a layered container image, a, which embeds an RPM and a Go executable.
+    #
+    # First, a parent image, p:
+    #   (p:Artifact) <-- (:Build) <-- (:SourceLocation)
+
+    # Then the container itself, a:
+    #   (a:Artifact) <-- (:Build) <-- (:SourceLocation)
+    #   (a:Artifact) <-[:EMBEDS]- (p:Artifact)
+    #
+    # And let's have the parent image embed an RPM, glibc:
+    #   (p:Artifact) <-[:EMBEDS]- (rpm:Artifact) <-- (:Build) <-- (internal:SourceLocation)
+    #                <-- (upstream:SourceLocation) <-- (:Component)
+
+    # Parent image, p
+    container_filename = ('docker-image-sha256:' + '0' * 64 + '.x86_64.tar.gz')
+    p = Artifact(
```

Comment:
I would appreciate a little bit longer name then 1 character.
It is simper to read assertions in following lines,

Reply:
How does it break pdb? Anyway, will make them longer. I think writing Go has made me appreciate shorter variable names :)

Reply:
And doesn't break `(i)pdb` where `p` is the shorthand for the print expression.

Reply:
I wasn't able to break pdb this way.

Reply:
(Oops, sorry for the double comment, I didn't realize it submitted it the first time.)

```
$ cat pdb_test.py 
from ipdb import set_trace

p = 'hello'
parent = 'world'
set_trace()
$ python pdb_test.py 
3.6.6 (default, Jul 19 2018, 14:25:17)
--Return--
None
> ./pdb_test.py(5)<module>()
      3 p = 'hello'
      4 parent = 'world'
----> 5 set_trace()

ipdb> parent
'world'
ipdb> p
*** SyntaxError: unexpected EOF while parsing
ipdb> p p
'hello'
```